### PR TITLE
With lib is bad

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/ulogd.nix
Comment (figsoda) [inline nixos/tests/ulogd.nix:4]: that's one use case of `with`  that I think makes sense, the scope is small enough imo

Comment (AndersonTorres) [reply nixos/tests/ulogd.nix:4]: It is insane to write

```
meta = with lib; {
  maintainers = with maintainers; [ just-one-person ];
}
```

instead of

```
meta.maintainers = [ lib.maintainers.just-one-person ];
```

Further, these codes are old, and for years no one appeared saying "hey, I wanna help maintaining this module, I will add myself to maintainers list!"

There is no justification to "accomodate the code for future changes".

Effectively nothing is being lost.

File: nixos/modules/testing/test-instrumentation.nix
Comment (AndersonTorres) [inline nixos/modules/testing/test-instrumentation.nix:121]: This is an example I will not touch. It is ugly but way less injustifiable than `meta = with lib; { maintainers = with pkgs.lib; . . .`

File: nixos/tests/ulogd.nix
Comment (figsoda) [reply nixos/tests/ulogd.nix:4]: > It is insane to write `...` instead of `...`

No, it is not.

> There is no justification to "accomodate the code for future changes".

There is. There are no qualifications for whether a module is "old enough" to need to switch from `with maintainers;` to `[ maintainers.<name ];`.

Additionally, if your argument is that this module isn't getting more maintenance, what is the point of proactively making this change then?

Comment (AndersonTorres) [reply nixos/tests/ulogd.nix:4]: > No, it is not.

The semantical quirks of nested `with` are known enough.

> There is. There are no qualifications for whether a module is "old enough" to need to switch from with maintainers; to [ maintainers.<name> ];.

In the best case, it justifies only a single `with lib.maintainers`, not a ridiculous `with lib; with maintainers;`.

> what is the point of proactively making this change in the first place?

Uniformization.

Further, beginners (and veterans too) tend to look for existing code as template for future code.

File: nixos/tests/systemd-initrd-networkd-ssh.nix
Comment (AndersonTorres) [inline nixos/tests/systemd-initrd-networkd-ssh.nix:3]: I swear I didn't change this!

**Comment by RaitoBezarius** [comment]
Please do not merge this without enough approvals. I will convert into draft so that no one merge it by mistake in the meantime.

**Comment by AndersonTorres** [comment]
> Please do not merge this without enough approvals. I will convert into draft so that no one merge it by mistake in the meantime.

I will not merge more than 70 commits, to not be so hard to review.

File: nixos/modules/testing/test-instrumentation.nix
Comment (mweinelt) [inline nixos/modules/testing/test-instrumentation.nix:7]:
```suggestion
  qemu-common = import ../../lib/qemu-common.nix { inherit lib pkgs; };
  
  inherit (lib)
    mkOverride
    ;
```

File: nixos/tests/binary-cache.nix
Comment (mweinelt) [inline nixos/tests/binary-cache.nix:11]: Having to qualify each list element individually does not look like an improvement to me.

File: nixos/tests/gitlab.nix
Comment (mweinelt) [inline nixos/tests/gitlab.nix:42]: inherit mkForce from lib

File: nixos/tests/pgjwt.nix
Comment (mweinelt) [inline nixos/tests/pgjwt.nix:10]:
```suggestion
        extraPlugins = with pkgs; [ pgjwt pgtap ];
```

File: nixos/tests/gitlab.nix
Comment (AndersonTorres) [reply nixos/tests/gitlab.nix:42]: Bikeshedding Battle.

I wanna invoke @supersandro2000.

What is better?

- add the `lib.` prefix in each `lib` function 

- inherit them on the top of file (or, for that matter, in any place before they are used)

File: nixos/tests/initrd-network-ssh/default.nix
Comment (SuperSandro2000) [inline nixos/tests/initrd-network-ssh/default.nix:47]: I would revert the formatting of this specific hunk as it is worse afterwards.

File: nixos/modules/config/sysctl.nix
Comment (SuperSandro2000) [inline nixos/modules/config/sysctl.nix:20]: I agree with most other changes but adding those long lists which need to maintained is IMO not an improvement and much more the opposite. I will always prefer writing lib in front of a function over maintaining this list. I would suggest to leave this file as it is to unblock all other changes where the improvement is clearly visible and not as discutable IMO.

File: nixos/tests/gitlab.nix
Comment (SuperSandro2000) [reply nixos/tests/gitlab.nix:42]: Prefixing lib here is totally fine IMO and better than maintaining an inherit list. In question just remove this change and we can merge the other things.

**Comment by AndersonTorres** [comment]
Removed all polemical commits.

File: nixos/tests/initrd-network-ssh/default.nix
Comment (AndersonTorres) [reply nixos/tests/initrd-network-ssh/default.nix:47]: Ok, I'll fix this later.

**Review by SuperSandro2000 — APPROVED** [review]
Lets get this merged and leave all the more discussion heavy changes to later.

**Comment by RaitoBezarius** [comment]
This was merged with one approval, this is not what I meant by "enough approvals", *sigh*.

**Comment by AndersonTorres** [comment]
The only "blocker" for the approvals was two convoluted patches and one disagreement between explicit lib prefix vs inherit listing. The overall intention was agreed.

**Comment by SuperSandro2000** [comment]
and I proposed we just remove the controversial changes and merge the things we agree on to move things forward.

**Comment by ajs124** [comment]
this broke the apparmor test. fix in #233850

**Pull request merged**
